src/views/bibliography_element.tsx

```diff
@@ -0,0 +1,254 @@
+    const dataTrackedMap: Map<string, TrackedAttrs> = new Map()
+
+    this.node.descendants((node, pos) => {
+      if (commentsDecorationSet) {
+        const nodePosition = this.getPos() + pos + 2
+        const commentWidget = commentsDecorationSet.find(
+          nodePosition,
+          nodePosition
+        )
+        if (commentWidget.length) {
+          const commentElement = (
+            commentWidget[0] as WidgetDecoration
+          ).type.toDOM()
+
+          commentElementMap.set(commentElement.id, commentElement)
+        }
+      }
+
+      const dataTracked = node.attrs.dataTracked as TrackedAttrs[]
+
+      if (dataTracked?.length) {
+        const lastChange = dataTracked[dataTracked.length - 1]
+        dataTrackedMap.set(node.attrs.id, lastChange)
+      }
+    })
+
+    const bibliographyFragment = await createBibliography(
+      bibliographyItems,
+      this.props.cslProps
+    )
+    const bibItems = bibliographyFragment.querySelectorAll('.bib-item')
+
+    bibItems.forEach((element) => {
+      const doubleButton = document.createElement('div')
+      const editButton = document.createElement('button')
+      const commentButton = document.createElement('button')
+
+      doubleButton.className = 'bibliography-double-button'
+      editButton.className = 'bibliography-edit-button'
+      commentButton.className = 'bibliography-comment-button'
+
+      commentButton.addEventListener('click', (e) => {
+        e.preventDefault()
+        this.props.setComment(buildComment(element.id) as CommentAnnotation)
+      })
+
+      editButton.addEventListener('click', (e) => {
+        e.preventDefault()
+        this.showPopper(element.id)
+        this.popperContainer = undefined
+      })
+
+      editButton.innerHTML = editIcon
+      commentButton.innerHTML = commentIcon
+      doubleButton.append(editButton, commentButton)
+
+      if (
+        this.props.getCapabilities().seeReferencesButtons &&
+        !element.querySelector('.bibliography-double-button')
+      ) {
+        element.appendChild(doubleButton)
+      }
+
+      editButton.disabled = !this.props.getCapabilities().editCitationsAndRefs
+
+      const commentElement = commentElementMap.get(element.id)
+
+      if (commentElement) {
+        element.childNodes[0].appendChild(commentElement)
+      }
+
+      const dataTracked = dataTrackedMap.get(element.id)
+
+      if (dataTracked) {
+        element.classList.add('attrs-track-mark')
+        const decoration = getMarkDecoration(dataTracked)
+
+        decoration.style && element.setAttribute('style', decoration.style)
+
+        if (
+          dataTracked.status === CHANGE_STATUS.pending &&
+          dataTracked.operation === CHANGE_OPERATION.set_node_attributes
+        ) {
+          element.childNodes[0].appendChild(
+            getAttrsTrackingButton(dataTracked.id)
+          )
+        }
+      }
+    })
+
+    const oldContent = this.container.querySelector('.contents')
```

Comment:
Would that be possible to use a variable to contain "contents" and use it here and on line 73?

Reply:
Yes, it's possible, but it requires passing the variable to the `createBibliography` function for use in line 73. The same applies to the `bib-item` class. Do you still feel it's a neater adjustment?

Reply:
My only concern here that some might change the selector in one place without changing it here but it's ok,